sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPlugin.scala

```diff
@@ -217,6 +217,18 @@ object ScalaJSPlugin extends AutoPlugin {
         "`scalaJSUseMainModuleInitializer` is true",
         CTask)
 
+    val scalaJSLinkerConfig = SettingKey[StandardLinker.Config](
+        "scalaJSLinkerConfig",
+        "Configuration of the Scala.js linker",
+        BPlusSetting)
+
+    val scalaJSLinkerBatchMode = SettingKey[Boolean](
+        "scalaJSLinkerBatchMode",
+        "Whether to run the Scala.js linker in batch mode, where nothing is " +
+        "cached nor incremental. This reduces the memory footprint, which " +
+        "can be particularly useful in a CI environment.",
+        BMinusSetting)
```

Comment:
I'm still unsure about this one. I kind of like that linkers don't have to worry about batch mode. But it's also disappointing that we have everything but one option in `scalaJSLinkerConfig`.

We could put the option inside `StandardLinker.Config`, and we might in the future enhance the linker to actually get rid of its caches more eagerly, e.g., throw out the optimizer as soon as it has done its job, rather than waiting for `link` to be completely finished.